### PR TITLE
e2e: kubectl HTTP fix for #15713 and #15787.

### PR DESCRIPTION
**Pull request by ashcrow**
This reverts commit 54c899370ee942fdda646bf3e199cc9b97cb9fb0, reversing
changes made to 0338e0ef5383037dcba5807ac8047c6b306e14c3 and adds a fix
proposed in #15848.

cc @jlowdermilk @fabioy

### CODE REVIEW

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -206,33 +206,53 @@ var _ = Describe("Kubectl client", func() {
...
+			apiServerUrl.Scheme = "https"
+			apiServerUrl.Path = "/api"
+			if !strings.Contains(apiServer, ":443") {
+				apiServerUrl.Host = apiServerUrl.Host + ":443"
+			}
+			apiServer = apiServerUrl.String()
 
 			// Get the kube/config
-			// TODO: Can it be RepoRoot with jenkins e2e?
 			testWorkspace := os.Getenv("WORKSPACE")
```
Comment (jlowdermilk) [inline test/e2e/kubectl.go]: This should be unnecessary now.

**Comment by ashcrow** [comment]
I have a feeling Shippable is going to flake. It's been "waiting" for over an hour.

File: test/e2e/kubectl.go [hunk 2]
```diff
@@ -206,33 +206,44 @@ var _ = Describe("Kubectl client", func() {
...
 			}
+
+			// Make sure the apiServer is set to what kubectl requires
 			apiServer := testContext.Host
-			// If there is no api in URL try to add it
-			if !strings.Contains(apiServer, ":443/api") {
-				apiServer = apiServer + ":443/api"
+			apiServerUrl, err := url.Parse(apiServer)
+			if err != nil {
+				Failf("Unable to parse URL %s. Error=Ts", apiServer, err)
```
Comment (ixdy) [inline test/e2e/kubectl.go]: "Error=Ts" looks like a typo?

**Comment by ashcrow** [comment]
Shippable failure is a flake (timeout).

**Comment by jlowdermilk** [comment]
lgtm. Shippable is currently in a bad way so I won't kick it. Trying to resolve.

**Comment by ashcrow** [comment]
@jlowdermilk it looks like Shippable is working again. I looked at a few recent PR's and *seems* to be as stable as it normally is.

**Comment by ashcrow** [comment]
@jlowdermilk \o/

**Pull request merged**